core/trino-main/src/main/java/io/trino/execution/RenameMaterializedViewTask.java

```diff
@@ -0,0 +1,108 @@
+import static io.trino.spi.StandardErrorCode.TABLE_ALREADY_EXISTS;
+import static io.trino.spi.StandardErrorCode.TABLE_NOT_FOUND;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+
+public class RenameMaterializedViewTask
+        implements DataDefinitionTask<RenameMaterializedView>
+{
+    @Override
+    public String getName()
+    {
+        return "RENAME MATERIALIZED VIEW";
+    }
+
+    @Override
+    public ListenableFuture<Void> execute(
+            RenameMaterializedView statement,
+            TransactionManager transactionManager,
+            Metadata metadata,
+            AccessControl accessControl,
+            QueryStateMachine stateMachine,
+            List<Expression> parameters,
+            WarningCollector warningCollector)
+    {
+        Session session = stateMachine.getSession();
+        QualifiedObjectName materializedViewName = createQualifiedObjectName(session, statement, statement.getSource());
+        Optional<ConnectorMaterializedViewDefinition> materializedView = metadata.getMaterializedView(session, materializedViewName);
+        if (materializedView.isEmpty()) {
+            Optional<ConnectorViewDefinition> view = metadata.getView(session, materializedViewName);
+            if (view.isPresent()) {
+                if (!statement.isExists()) {
+                    throw semanticException(
+                            TABLE_NOT_FOUND,
+                            statement,
+                            "Materialized View '%s' does not exist, but a view with that name exists. Did you mean ALTER VIEW %s RENAME ...?", materializedViewName, materializedViewName);
+                }
+                return immediateVoidFuture();
+            }
+
+            Optional<TableHandle> table = metadata.getTableHandle(session, materializedViewName);
+            if (table.isPresent()) {
+                if (!statement.isExists()) {
+                    throw semanticException(
+                            TABLE_NOT_FOUND,
+                            statement,
+                            "Materialized View '%s' does not exist, but a table with that name exists. Did you mean ALTER TABLE %s RENAME ...?", materializedViewName, materializedViewName);
+                }
+                return immediateVoidFuture();
+            }
+
+            if (statement.isExists()) {
+                return immediateVoidFuture();
+            }
+            throw semanticException(TABLE_NOT_FOUND, statement, "Materialized View '%s' does not exist", materializedViewName);
+        }
+
+        QualifiedObjectName target = createQualifiedObjectName(session, statement, statement.getTarget());
+        if (metadata.getCatalogHandle(session, target.getCatalogName()).isEmpty()) {
+            throw semanticException(CATALOG_NOT_FOUND, statement, "Target catalog '%s' does not exist", target.getCatalogName());
+        }
+        if (metadata.getMaterializedView(session, target).isPresent()) {
```

Comment:
nit: It seems that would try a rename if table or view with same name exist

Reply:
indeed we would. This is consistent with rename on a view or table.
I will add the checks here. It would be good to add it to the view and table rename tasks as well.